src/tainting/Dataflow_tainting.ml

```diff
@@ -998,76 +998,118 @@ and check_tainted_expr env exp : Taints.t * Lval_env.t =
 let check_tainted_var env (var : IL.name) : Taints.t * Lval_env.t =
   check_tainted_lval env (LV.lval_of_var var)
 
+let lval_of_sig_arg fun_exp fparams args_exps (sig_arg : T.arg) =
```

Comment:
This function took me a long time to read. I went back and read the whole PR, then came back, and that helped, but it's still difficult because the context is far away. Some comments might be helpful. For instance, what is a `sig_arg`?